### PR TITLE
Updating Style Guide Headings and Titles section

### PR DESCRIPTION
**Pull request by nate-double-u**
### Description

Updates the style guide "Headings and titles" section with guidance on:
1. links in headers
2. using properly formatted headers

[Deploy preview](https://deploy-preview-48667--kubernetes-io-main-staging.netlify.app/docs/contribute/style/style-guide/#headings)

### Issue

Closes: #48662

### CODE REVIEW

File: content/en/docs/contribute/style/style-guide.md [hunk 1]
```diff
@@ -510,6 +510,8 @@ Use ordered headings to provide a meaningful high-level outline of your content.
 Use pound or hash signs (`#`) for non-blog post content. | Use underlines (`---` or `===`) to designate first-level headings.
 Use sentence case for headings in the page body. For example, **Extend kubectl with plugins** | Use title case for headings in the page body. For example, **Extend Kubectl With Plugins**
 Use title case for the page title in the front matter. For example, `title: Kubernetes API Server Bypass Risks` | Use sentence case for page titles in the front matter. For example, don't use `title: Kubernetes API server bypass risks`
+Place relevant links in the body copy. | Include hyperlinks (`<a href=""></a>`) in headings.
```
Comment (ArvindParekh) [inline content/en/docs/contribute/style/style-guide.md:513]: Does "body copy" mean something in particular here? Shouldn't it be just "body" or "body below the Headings"?
Sorry, saw this as a recent contributor and found it a bit confusing, so maybe others might as well :)

Comment (nate-double-u) [reply content/en/docs/contribute/style/style-guide.md:513]: [Body copy or body text](https://en.wikipedia.org/wiki/Body_text) refers to regular text. In this case, it refers to a nearby paragraph. I don’t want to prescribe that it be above or below, as it depends on the situation.

Comment (ArvindParekh) [reply content/en/docs/contribute/style/style-guide.md:513]: Ah, didn't know that. That makes sense, thanks for sharing 👍🏻

**Review by dipesh-rawat — COMMENTED** [review]
Useful addition for the style guide.

/approve

**Pull request merged**